### ISSUE TITLE
`--curves` parameter should not be applied for DoH transfer

### ISSUE THREAD

**Issue by chengr28** [curl/curl#14578]
### I did this

`curl --doh-url "https://dns.google/dns-query" "https://www.google.com"`
`curl --doh-url "https://dns.google/dns-query" --curves "X25519" "https://www.google.com"`
`curl --doh-url "https://dns.google/dns-query" --curves "X25519:P-256" "https://www.google.com"`
`curl --doh-url "https://dns.google/dns-query" --curves "X25519:P-256:P-384" "https://www.google.com"`
`curl --doh-url "https://dns.google/dns-query" --curves "X25519:P-256:P-384:P-521" "https://www.google.com"`

### I expected the following

Hi team,

I'm testing DoH, found that `--curves` parameter is applied to DoH connections.
After tracing why some TLS options are not inherited from the user, I got `make DoH inherit more transfer properties` in TODO, and the PR https://github.com/curl/curl/pull/3661.
It seems TLS curves should also be one of them if TLS version/ciphers/.. options are not be applied for DoH transfer, or at least it should be controlled by a separately DoH-related parameter.

### curl/libcurl version

curl 8.9.1 (x86_64-w64-mingw32) libcurl/8.9.1 LibreSSL/3.9.2 zlib/1.3.1 brotli/1.1.0 zstd/1.5.6 WinIDN libpsl/0.21.5 libssh2/1.11.0 nghttp2/1.62.1 ngtcp2/1.6.0 nghttp3/1.4.0
Release-Date: 2024-07-31
Protocols: dict file ftp ftps gopher gophers http https imap imaps ipfs ipns ldap ldaps mqtt pop3 pop3s rtsp scp sftp smb smbs smtp smtps telnet tftp ws wss
Features: alt-svc AsynchDNS brotli HSTS HTTP2 HTTP3 HTTPS-proxy IDN IPv6 Kerberos Largefile libz NTLM PSL SPNEGO SSL SSPI threadsafe UnixSockets zstd

### operating system

Windows 11 23H2

**Comment by bagder** [comment]
I believe we have hesitated a bit to explode the number of options before we know this is actually something people *want*. It sounds to me that even you don't really *need* this, but mostly found this because you were testing out how it works?

**Comment by chengr28** [comment]
Hi bagder,

Yes, you are right, it just a test.

In the beginning, I tried `--ech`, unfortunately it doesn't work on Windows. DoH was also tested (`--ech true` requires DoH), then I found this.

Before starting the test, I think all TLS parameters will be inherited from my command until packet capture showing different result. PR https://github.com/curl/curl/pull/3661 explains why some TLS options are not applied for DoH transfer, except TLS curves (`--curves` seems to be added after supporting DoH, in commit https://github.com/curl/curl/commit/ede125b7b7ca8fc5a1fe3d7c1aee6bff2ea0bf24 ).

That's why I create this issue. In my opinion, curves is one of TLS connection related parameter, and should also be included in the exception list (TLS version, ciphers, etc.). That makes me confused once I found `--curves` is applied for DoH transfer.

I believe the team is still collecting advise of DoH part, so I think that these parameters should keep the same behavior before deciding to inherit or add new options. For example, if TLS ciphers should (not) be inherited from user, then TLS curves should do the same.

By the way, copying the existing TLS options to a new protocol may not a good idea, that will makes the number of option double.
